src/main/java/net/coderbot/iris/UpdateChecker.java

```diff
@@ -0,0 +1,120 @@
+package net.coderbot.iris;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonParser;
+import net.coderbot.iris.config.IrisConfig;
+import net.coderbot.iris.gl.shader.StandardMacros;
+import net.fabricmc.loader.api.FabricLoader;
+import net.minecraft.client.Minecraft;
+import net.minecraft.network.chat.ClickEvent;
+import net.minecraft.network.chat.Component;
+import net.minecraft.network.chat.MutableComponent;
+import net.minecraft.network.chat.TextComponent;
+import org.apache.commons.io.FileUtils;
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang3.time.DateUtils;
+
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.util.Date;
+import java.util.Locale;
+import java.util.Objects;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+
+public class UpdateChecker {
+	private final int simpleVersion;
+	private CompletableFuture<UpdateInfo> info;
+	private boolean shouldShowUpdateMessage;
+	private boolean usedIrisInstaller;
+
+	public UpdateChecker(int simpleVersion) {
+		this.simpleVersion = simpleVersion;
+		if (Objects.equals(System.getProperty("iris.installer", "false"), "true")) {
+			usedIrisInstaller = true;
+		}
+	}
+
+	public void checkForUpdates(IrisConfig irisConfig) {
+		this.info = CompletableFuture.supplyAsync(() -> {
+			if (!irisConfig.shouldDisableUpdateMessage()) {
+				try {
+					File updateFile = FabricLoader.getInstance().getGameDir().resolve("irisUpdateInfo.json").toFile();
+					if (DateUtils.isSameDay(new Date(), new Date(updateFile.lastModified()))) {
+						Iris.logger.warn("Cached update file detected, using that!");
+						UpdateInfo updateInfo = new Gson().fromJson(FileUtils.readFileToString(updateFile, StandardCharsets.UTF_8), UpdateInfo.class);
+						if (updateInfo.simpleVersion > simpleVersion) {
```

Comment:
I'm not sure if the simpleVersion stuff is the right approach. This won't show an update warning for someone using `1.2.5-rc.1` if `1.2.5` is the latest veresion. Fabric Loader has some good code for SemVer version comparisons that we probably want to use instead.